### ISSUE TITLE
Can it support M1 Mac ?

### ISSUE THREAD

**Issue by wastu01** [openai/shap-e#22]

**Comment by wastu01** [comment]
and I had this problem and solved by changing Wifi setting

[Pip error, OSError Errno 22 invalid argument](https://stackoverflow.com/questions/46942585/pip-error-oserror-errno-22-invalid-argument)

Python version: 3.11.2
OS: macOS Ventura 13.2 M1

**Comment by DutchMaxwell** [comment]
That would be awesome!

**Comment by xbwtyz** [comment]
No, get a new computer and stop asking people to support your bad decision.

**Comment by arul-sinha** [comment]
> No, get a new computer and stop asking people to support your bad decision.

Not a bad descision to support a apple silicon chip. So maybe u stop commenting on such things.

**Comment by xbwtyz** [comment]
Please explain why time should be wasted accommodating a minuscule portion of Apple's 8% market share. Why should every completed project have to waste time making it work for you? Why don't you learn how to use Xcode & Rosetta and do it yourself? 

I use Windows, but you don't see me in threads asking people to port to Docker or whatever VENV I prefer, I get off my rear and do it myself. Welcome to Github.

**Comment by arul-sinha** [comment]
> Please explain why time should be wasted accommodating a minuscule portion of Apple's 8% market share. Why should every completed project have to waste time making it work for you? Why don't you learn how to use Xcode & Rosetta and do it yourself?
> 
> I use Windows, but you don't see me in threads asking people to port to Docker or whatever VENV I prefer, I get off my rear and do it myself. Welcome to Github.

ur the one wasting time here by posting this stuff. Newbies exist bro. maybe ur just pissed because you cant afford a mac. dont take it out here

**Comment by yoginsurya** [comment]
> Please explain why time should be wasted accommodating a minuscule portion of Apple's 8% market share. Why should every completed project have to waste time making it work for you? Why don't you learn how to use Xcode & Rosetta and do it yourself?
> 
> I use Windows, but you don't see me in threads asking people to port to Docker or whatever VENV I prefer, I get off my rear and do it myself. Welcome to Github.

Respectfully, here's the explanation:

1. Market share is not a great proxy for your argument. This is a dev heavy website, but you're using a % which represents the global population. Look at the stack overflow developer survey and you'll see ~32% share of **devs** using it for personal/work.

2. Ollama is one of the **most popular** ways to run LLMs locally, and this _was_ one of its most requested features (Windows support): https://github.com/ollama/ollama/issues/403

3. "I get off my rear and do it myself. Welcome to Github." -- you have zero contributions in your profile; maybe this is a dummy profile, who knows. Maybe, through that same logic, you should have got off your rear and created the code for Shap-E yourself?

Open-source is about sharing contributions and building on others' work. Requesting features helps devs gauge interest—OP did so respectfully.
